src/components/utils/CalciteExpandToggle.tsx

```diff
@@ -0,0 +1,67 @@
+import { FunctionalComponent, h } from "@stencil/core";
+import { getElementDir } from "./dom";
+import { CalciteLayout } from "../interfaces";
+
+interface CalciteExpandToggleProps {
+  expanded: boolean;
+  textExpand: string;
+  textCollapse: string;
+  el: HTMLElement;
+  layout: CalciteLayout;
+  toggleExpand: () => void;
+}
+
+const ICONS = {
+  chevronsLeft: "chevrons-left",
+  chevronsRight: "chevrons-right"
+};
+
+function getClosestShellLayout(el: HTMLElement): CalciteLayout {
```

Comment:
Should this be` CalciteLayout | null` to make it explicit that null may be returned?